tensorflow/core/kernels/whole_file_read_ops.cc

```diff
@@ -119,4 +119,34 @@ class ReadFileOp : public OpKernel {
 
 REGISTER_KERNEL_BUILDER(Name("ReadFile").Device(DEVICE_CPU), ReadFileOp);
 
+class WriteFileOp : public OpKernel {
+  public:
+    using OpKernel::OpKernel;
+    void Compute(OpKernelContext* context) override {
+	    const Tensor* filename_input;
+	    const Tensor* contents_input;
+	    Tensor* output = nullptr;
+	    OP_REQUIRES_OK(context, context->input("filename", &filename_input));
+	    OP_REQUIRES_OK(context, context->input("contents", &contents_input));
+	    OP_REQUIRES(context, TensorShapeUtils::IsScalar(filename_input->shape()),
+			    errors::InvalidArgument(
+				    "Input filename tensor must be scalar, but had shape: ",
+				    filename_input->shape().DebugString()));
+	    OP_REQUIRES(context, TensorShapeUtils::IsScalar(contents_input->shape()),
+			    errors::InvalidArgument(
+				    "Contents tensor must be scalar, but had shape: ",
+				    contents_input->shape().DebugString()));
+	    OP_REQUIRES_OK(context, context->allocate_output("output", TensorShape({}),
+				    &output));
+	    if(WriteStringToFile(context->env(),
+				    filename_input->scalar<string>()(),
+				    contents_input->scalar<string>()()).ok()) {
+	    	output->scalar<int>()(0) = 0;
+	    } else {
+	    	output->scalar<int>()(0) = -1;
```

Comment:
On second thought, I think it'll be less burden on user of this Op just throw an error if the write is unsuccessful.  (Otherwise every responsible caller of this Op need to do something like `tf.assert_equal(tf.write_file(...), 0)`.)

Let's get rid of the output, and do `OP_REQUIRES_OK(context, WriteStringToFile(...))` directly here (when it fails, a nice stack trace will be printed out).